src/Workspaces/Core/Portable/Workspace/Solution/ChecksumsAndIds.cs

```diff
@@ -78,3 +79,63 @@ public bool MoveNext()
             => (_checksumsAndIds.Checksums.Children[_index], _checksumsAndIds.Ids[_index]);
     }
 }
+
+internal readonly struct DocumentChecksumsAndIds
```

Comment:
>DocumentChecksumsAndIds [](http://example.com/codeflow?start=25&length=23)

Seems like this would replace ChecksumsAndIds<DocumentId>, but it must not as that is still used in this branch.

Reply:
Maybe the former is still just there in use cases where the caller doesn't need all the data?